internal/pkg/service/stream/api/service/sink.go

```diff
@@ -62,13 +62,13 @@ func (s *service) CreateSink(ctx context.Context, d dependencies.SourceRequestSc
 			// Create sink
 			op := s.definition.Sink().Create(&sink, s.clock.Now(), d.RequestUser(), "New sink.")
 			op = op.RequireLock(lock)
-			if err := op.Do(ctx).Err(); err == nil {
-				result := task.OkResult("Sink has been created successfully.")
-				result = s.mapper.WithTaskOutputs(result, sink.SinkKey)
-				return result
-			} else {
+			if err := op.Do(ctx).Err(); err != nil {
 				return task.ErrResult(err)
 			}
+
+			result := task.OkResult("Sink has been created successfully.")
+			result = s.mapper.WithTaskOutputs(result, sink.SinkKey)
+			return result
```

Comment:
In my opinion the intended (happy) path should be left-aligned and errors should be handled in the ifs so I swapped the branches accordingly and removed unnecessary else branches.